From d6576b798868a84e4a6e347743f691b444bb1dc6 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 6 Apr 2023 13:09:15 -0700
Subject: [PATCH 01/10] SUBMITTED: MIPS: generic: Do not select CPUs that are
 unsupported in clang

When building allnoconfig with clang after commit de34007751aa ("MIPS:
generic: Enable all CPUs supported by virt board in Kconfig"), the
following error occurs:

  error: unknown target CPU 'r4600'
  note: valid target CPU values are: mips1, mips2, mips3, mips4, mips5, mips32, mips32r2, mips32r3, mips32r5, mips32r6, mips64, mips64r2, mips64r3, mips64r5, mips64r6, octeon, octeon+, p5600

Working around that, there are similar errors for 'loongson2e' and
'loongson2f'.

These CPUs are not supported in clang/LLVM, so do not select support for
them in MIPS_GENERIC_KERNEL when building with clang.

Fixes: de34007751aa ("MIPS: generic: Enable all CPUs supported by virt board in Kconfig")
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20230406-mips-clang-generic-selects-fix-v1-1-811690c9fb69@kernel.org
---
 arch/mips/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 770d098b11bf..12aed9e12704 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -152,15 +152,15 @@ config MIPS_GENERIC_KERNEL
 	select SMP_UP if SMP
 	select SWAP_IO_SPACE
 	select SYS_HAS_CPU_CAVIUM_OCTEON
-	select SYS_HAS_CPU_LOONGSON2E
-	select SYS_HAS_CPU_LOONGSON2F
+	select SYS_HAS_CPU_LOONGSON2E if !CC_IS_CLANG
+	select SYS_HAS_CPU_LOONGSON2F if !CC_IS_CLANG
 	select SYS_HAS_CPU_MIPS32_R1
 	select SYS_HAS_CPU_MIPS32_R2
 	select SYS_HAS_CPU_MIPS32_R6
 	select SYS_HAS_CPU_MIPS64_R1
 	select SYS_HAS_CPU_MIPS64_R2
 	select SYS_HAS_CPU_MIPS64_R6
-	select SYS_HAS_CPU_R4X00
+	select SYS_HAS_CPU_R4X00 if !CC_IS_CLANG
 	select SYS_SUPPORTS_32BIT_KERNEL
 	select SYS_SUPPORTS_64BIT_KERNEL
 	select SYS_SUPPORTS_BIG_ENDIAN

base-commit: e134c93f788fb93fd6a3ec3af9af850a2048c7e6
-- 
2.40.0

